lib/src/working_copy.rs

```diff
@@ -398,6 +402,12 @@ pub enum TreeStateError {
         path: PathBuf,
         source: tempfile::PersistError,
     },
+    #[cfg(feature = "watchman")]
+    #[error("Initializing Watchman filesystem monitor: {0}")]
+    InitWatchman(watchman::Error),
+    #[cfg(feature = "watchman")]
+    #[error("Querying Watchman filesystem monitor: {0}")]
+    QueryWatchman(watchman::Error),
```

Comment:
I'm not sure if we'll add another fsmonitor backend, but maybe better to add a generic error kind like `FsMonitor(Box<dyn Error + Send + Sync>)`?